File: common-gui/src/main/java/ru/usedesk/common_gui/internal/di/InstanceBox.java
package ru.usedesk.common_gui.internal.di;

import android.support.annotation.NonNull;

import javax.inject.Inject;

import ru.usedesk.common_gui.external.IUsedeskViewCustomizer;
import ru.usedesk.common_sdk.internal.appdi.InjectBox;

public class InstanceBox extends InjectBox {
    @Inject
    IUsedeskViewCustomizer usedeskViewCustomizer;

    public InstanceBox() {
        init(new MainModule());
    }

    @NonNull
    public IUsedeskViewCustomizer getUsedeskViewCustomizer() {
        return usedeskViewCustomizer;
    }
}

File: knowledgebase-sdk/src/main/java/ru/usedesk/knowledgebase_sdk/di/InstanceBox.kt
package ru.usedesk.knowledgebase_sdk.di

import android.content.Context
import ru.usedesk.common_sdk.di.InjectBox
import ru.usedesk.knowledgebase_sdk.domain.IUsedeskKnowledgeBase
import ru.usedesk.knowledgebase_sdk.entity.UsedeskKnowledgeBaseConfiguration
import toothpick.ktp.delegate.inject

internal class InstanceBox(
        appContext: Context, knowledgeBaseConfiguration:
        UsedeskKnowledgeBaseConfiguration
) : InjectBox() {

    val knowledgeBaseSdk: IUsedeskKnowledgeBase by inject()

    init {
        init(MainModule(appContext, knowledgeBaseConfiguration))
    }
}